src/spec-lang/tc/semcheck.ts

```diff
@@ -125,7 +125,16 @@ export function scAnnotation(
         }
         sc(node.expression, ctx, typings, semMap);
     } else if (node instanceof SUserConstantDefinition) {
-        sc(node.value, ctx, typings, semMap);
+        const info = sc(node.value, ctx, typings, semMap);
+
+        if (!info.isConst) {
+            throw new SemError(
+                `Cannot use non-constant expression ${node.value.pp()} in constant definition for ${
+                    node.name
```

Comment:
Shouldn't here be `node.name.name` or `node.name.pp()` instead of `node.name` (which is probably an `SId`)?

Reply:
Fixed in [1f5c2ca](https://github.com/ConsenSys/scribble/pull/180/commits/1f5c2cafd2d5c6dce1b8dad70df3ae3515e1abad)